searchlib/src/vespa/searchlib/common/bitvector.h

```diff
@@ -189,14 +189,16 @@ class BitVector : protected BitWord
      * should be called before calling Test/Clear/Flip methods.
      */
     void invalidateCachedCount() const {
-        _numTrueBits = invalidCount();
+        _numTrueBits.store(invalidCount(), std::memory_order_relaxed);
     }
 
     void swap(BitVector & rhs) {
         std::swap(_words, rhs._words);
         std::swap(_startOffset, rhs._startOffset);
         std::swap(_sz, rhs._sz);
-        std::swap(_numTrueBits, rhs._numTrueBits);
+        Index tmp = rhs._numTrueBits;
+        rhs._numTrueBits = _numTrueBits.load();
+        _numTrueBits = tmp;
```

Comment:
Consider adding `load()`/`store()` to these lines where appropriate to make all atomic access explicit (if desired)

Reply:
Fixed